src/App.tsx

```diff
@@ -5,17 +5,86 @@ import { Counter } from './components/Counter/Counter';
 import { MovieTile } from './components/MovieTile/MovieTile';
 import { MovieDetails } from './components/MovieDetails/MovieDetails';
 import { SortControl } from './components/SortControl/SortControl';
-import { DetailsContainer, MoviesContainer, DetailsHeader } from './App.styled';
+import {
+  DetailsContainer,
+  MoviesContainer,
+  DetailsHeader,
+  ConfirmButton,
```

Comment:
Please check unused imports.
Or please check the comment below. Seems like you wanted to pass dialog content from here.
This was good option. Instead of checking the type of children.

Another option can be creation of Default dialog with juts children and two or three more dialogs based on default one. One for edit and one for delete. 

But here the first option will be enough. Just one Dialog and dinamyc content where you use this dialog.